nx/source/services/hid.c

```diff
@@ -49,8 +49,9 @@ Result hidInitialize(void)
     u64 AppletResourceUserId;
 
     rc = appletGetAppletResourceUserId(&AppletResourceUserId);
+    // If this failed (for example because we're a sysmodule) AppletResourceUserId stays zero
     if (R_FAILED(rc))
-        return rc;
+        AppletResourceUserId = 0;
```

Comment:
Does hidSetNpadJoyAssignmentModeDual() work with this same change if the below check is removed?

Reply:
Yeah, it does!

I'll update my pr with better changes.